maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/AbstractSurefireMojo.java

```diff
@@ -197,7 +197,7 @@
      * The base directory of the project being tested. This can be obtained in your integration test via
      * System.getProperty("basedir").
      */
-    @Parameter( defaultValue = "${basedir}" )
+    @Parameter( defaultValue = "${basedir}", readonly = true )
```

Comment:
I checked the history in 2012 and this was `required` but not `readonly`.
IMHO we should keep it and put `required` to `testClassesDirectory`.